src/Traits/ErrorMessageDetectionTrait.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PNX\DrupalTestUtils\Traits;
+
+use Drupal\Component\Utility\Html;
+
+/**
+ * Helper functions to detect if the response output contains an error.
+ */
+trait ErrorMessageDetectionTrait {
+
+  /**
+   * Detects an error in the response output and fails with the error message.
+   */
+  protected function detectErrorMessageInResponseOutput(string $response): void {
+    $error = 'The website encountered an unexpected error. Please try again later.';
```

Comment:
Note, this won't detect Fatal Error messages, e.g. OOM. Maybe we could expand this a little.